File: src/main/java/es/upm/dit/isst/mystayapi/model/Reserva.java
package es.upm.dit.isst.mystayapi.model;

import java.sql.Time;

import com.fasterxml.jackson.annotation.JsonProperty;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
public class Reserva {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idreserva")
    private Integer id;

    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "idcliente", nullable = false)
    @Column(nullable = false, unique=true)
    private String Cliente_id;

    private Time Fechainicio;

    private Time Fechafinal;

    private Integer Llave;
    
    private String Hotel;



}

File: mystay-api/models/reserva.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Reserva struct {
	gorm.Model
	ID          int `gorm:"primary_key"`
	ClienteID   int
	Client      Client `gorm:"foreignKey:ClienteID"`
	FechaInicio time.Time
	FechaFinal  time.Time
	Llave       string
	HotelID     int
}
